node/rest/src/helpers/auth.rs

```diff
@@ -80,16 +84,20 @@ pub async fn auth_middleware(request: Request<Body>, next: Next) -> Result<Respo
     let auth: TypedHeader<Authorization<Bearer>> =
         parts.extract().await.map_err(|_| StatusCode::UNAUTHORIZED.into_response())?;
 
-    match decode::<Claims>(auth.token(), &DecodingKey::from_secret(jwt_secret()), &Validation::new(Algorithm::HS256)) {
+    match decode::<Claims>(
+        auth.token(),
+        &DecodingKey::from_secret(JWT_SECRET.get().unwrap()),
+        &Validation::new(Algorithm::HS256),
+    ) {
         Ok(decoded) => {
             let claims = decoded.claims;
             if claims.is_expired() {
                 return Err((StatusCode::UNAUTHORIZED, "Expired JSON Web Token".to_owned()).into_response());
             }
         }
 
-        Err(_) => {
-            return Err(StatusCode::UNAUTHORIZED.into_response());
+        Err(err) => {
```

Comment:
Hate to do one more nit, but I've seen it being good security practice not to return  internal errors to potential attackers. What about logging this locally instead of returning it?

Reply:
That's reasonable; I'll make it into a `WARN`-tier log, perhaps, so that the server can have greater visibility in case of brute-force access attempts.